cli/src/main/java/org/aya/cli/parse/AyaProducer.java

```diff
@@ -366,11 +366,94 @@ public QualifiedID visitQualifiedId(AyaParser.QualifiedIdContext ctx) {
         sourcePosOf(forall), false,
         visitForallTelescope(forall.tele()).view(),
         visitExpr(forall.expr()));
+      case AyaParser.DoContext doCtx -> visitDo(doCtx);
+      case AyaParser.IdiomContext idmCtx -> {
+        if (idmCtx.idiomBlock().barredExpr() == null)
+          yield new Expr.UnresolvedExpr(sourcePosOf(idmCtx), "empty");
+        yield visitIdiomBlock(idmCtx.idiomBlock());
+      }
+      case AyaParser.ArrayContext arrCtx -> {
+        if (arrCtx.exprList() == null)
+          yield new Expr.UnresolvedExpr(sourcePosOf(arrCtx), "nil");
```

Comment:
Consider extracting constant strings to `Constants` class

Reply:
Will do.